nemo/collections/asr/models/asr_model.py

```diff
@@ -79,8 +79,12 @@ def input_module(self):
     def output_module(self):
         return self.decoder
 
-    def forward_for_export(self, input):
-        return self.output_module(self.input_module(input))
+    def forward_for_export(self, input, length=None):
+        encoder_output = self.input_module(input, length)
+        if isinstance(encoder_output, tuple):
```

Comment:
I believe, that would be more consistent if you modify forward() of the new encoder to return a single value if called with Len=None ? Or do you actually have a case where forward() does not receive length, but should return it anyway ?

Reply:
Actually in the new models like Conformer, we have sub-sampling which reduces the lengths, so we need to always return the updated lengths. In case of None for length, we assume the whole input is valid and produce the lengths automatically.  Due to the sub-sampling, the lengths get reduces, so need to return it. Older models did not change the lengths inside the encoder.

Reply:
OK! The change is good then.